### PR TITLE
fix `Script error on calling internal handlers for page containing 'beforeunload' handler` (close #1419)

### LINKED ISSUE #1419: [UnloadSandbox] Script error on calling internal handlers for page containing 'beforeunload' handler
Error:
```js
Uncaught TypeError: originListener is not a function
    at UnloadSandbox._onBeforeUnloadHandler (hammerhead.js:7154)
    at hammerhead.js:7174
    at hammerhead.js:5217
```

Code for reproduce:

#### index.html
```html
<html>
	<body>
		<form method="post" target="submitTarget">
			<input id="submitBtn" type="submit">
		</form>
		<iframe id="submitTarget" name="submitTarget" style="display: none"></iframe>
		<script>
			window.addEventListener('beforeunload', function (e) {
				window.test = true;
			});
		</script>
	</body>
</html>
```
#### index.js
```js
var http = require('http');
var fs   = require('fs');

http.createServer(function (req, res) {
    var content = '';

    if (req.url === '/')
        content =fs.readFileSync('index.html');
    else if (req.url === '/login')
        content = '<html><body><h1>Form response</h1></body></html>';

    res.end(content);
}).listen(3000);

```

### PR DESCRIPTION
**Pull request by LavrovArtem**

### CODE REVIEW

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

File: src/client/sandbox/node/document/index.js [hunk 1]
```diff
@@ -77,6 +77,20 @@ export default class DocumentSandbox extends SandboxBase {
 
             const result = nativeMethods.documentOpen.apply(document, args);
 
+            if (window[INTERNAL_PROPS.hammerhead]) {
+                window[INTERNAL_PROPS.hammerhead].nativeMethods.objectDefineProperty
+                    .call(window.Object, window, INTERNAL_PROPS.isDocumentWasCleaned, {
```
Comment (miherlosev) [inline src/client/sandbox/node/document/index.js]: rename to `INTERNAL_PROPS.documentWasCleaned`

File: src/client/sandbox/node/document/index.js [hunk 2]
```diff
@@ -77,6 +81,14 @@ export default class DocumentSandbox extends SandboxBase {
 
             const result = nativeMethods.documentOpen.apply(document, args);
 
+            // NOTE: Chrome does not remove the "%hammerhead%" property from window
+            const nativeObjectDefineProperty = window[INTERNAL_PROPS.hammerhead]
```
Comment (miherlosev) [inline src/client/sandbox/node/document/index.js]: `const objectDefinePropertyFn = `

File: src/client/sandbox/node/document/index.js [hunk 3]
```diff
@@ -77,6 +81,14 @@ export default class DocumentSandbox extends SandboxBase {
 
             const result = nativeMethods.documentOpen.apply(document, args);
 
+            // NOTE: Chrome does not remove the "%hammerhead%" property from window
```
Comment (miherlosev) [inline src/client/sandbox/node/document/index.js]: // NOTE: Chrome does not remove the "%hammerhead%" property from window
// after document.write call

File: src/client/utils/dom.js [hunk 4]
```diff
@@ -339,9 +339,13 @@ export function isIframeWithoutSrc (iframe) {
     const iframeSrcLocation      = iframeLocation.srcLocation;
     const iframeDocumentLocation = iframeLocation.documentLocation;
 
-    if (iframeDocumentLocation === null) // is a cross-domain iframe
+    // NOTE: is a cross-domain iframe
+    if (iframeDocumentLocation === null)
         return false;
 
+    if (iframe.contentWindow[INTERNAL_PROPS.documentWasCleaned])
```
Comment (miherlosev) [inline src/client/utils/dom.js]: Add comment
// NOTE:  after 'document.write ' call for iframe with/without  src we will process it as iframe without src

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by LavrovArtem** [comment]
ping @churkin

**Pull request merged**